config/config.go

```diff
@@ -444,6 +451,124 @@ func (c *SlackConf) Validate() (errs []error) {
 	return
 }
 
+// SyslogConf is syslog config
+type SyslogConf struct {
+	Protocol string
+	Host     string `valid:"host"`
+	Port     string `valid:"port"`
+	Severity string
+	Facility string
+	Tag      string
+
+	Verbose bool
+}
+
+// Validate validates configuration
+func (c *SyslogConf) Validate() (errs []error) {
+	//  If protocol is empty, it will connect to the local syslog server.
+	if len(c.Protocol) > 0 && c.Protocol != "tcp" && c.Protocol != "udp" {
+		errs = append(errs, errors.New(`protocol must be "tcp" or "udp"`))
+	}
+
+	// Default port: 514
+	if c.Port == "" {
+		c.Port = "514"
+	}
+
+	if _, err := c.GetSeverity(); err != nil {
+		errs = append(errs, err)
+	}
+
+	if _, err := c.GetFacility(); err != nil {
+		errs = append(errs, err)
+	}
+
+	if _, err := valid.ValidateStruct(c); err != nil {
+		errs = append(errs, err)
+	}
+	return errs
+}
+
+// GetSeverity gets severity
+func (c *SyslogConf) GetSeverity() (syslog.Priority, error) {
+	if c.Severity == "" {
+		return syslog.LOG_INFO, nil
+	}
+
+	switch c.Severity {
```

Comment:
Wouldn't a map associating string to the syslog constants be shorter and more readable?
Something like:
```go
if p, ok := priorities[c.Severity]; ok {
  return p, nil
}
return -1, fmt.Errorf("Invalid severity: %s", c.Severity)
```

Reply:
Whichever is OK :)